azuredevops/internal/service/approvals_and_checks/common_check.go

```diff
@@ -0,0 +1,241 @@
+package approvals_and_checks
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/pipelineschecks"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+var taskCheckType = pipelineschecks.CheckType{
+	Id: converter.UUID("fe1de3ee-a436-41b4-bb20-f6eb4cb879a7"),
+}
+
+// NOTE: In theory the API should accept "agentpool" as well, but the API client requires a project ID
+// so it doesn't seem to work and the website UI doesn't have it available
+var targetResourceTypes = []string{"endpoint", "environment", "queue", "repository", "securefile", "variablegroup"}
+
+type flatFunc func(d *schema.ResourceData, check *pipelineschecks.CheckConfiguration, projectID string) error
+type expandFunc func(d *schema.ResourceData) (*pipelineschecks.CheckConfiguration, string, error)
+
+// genBaseCheckResource creates a Resource with the common parts
+// that all checks require.
+func genBaseCheckResource(f flatFunc, e expandFunc) *schema.Resource {
+	return &schema.Resource{
+		Create: genCheckCreateFunc(f, e),
+		Read:   genCheckReadFunc(f),
+		Update: genCheckUpdateFunc(f, e),
+		Delete: genCheckDeleteFunc(e),
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(2 * time.Minute),
+			Read:   schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(2 * time.Minute),
+			Delete: schema.DefaultTimeout(2 * time.Minute),
+		},
+		Importer: nil,
```

Comment:
Can be remove if `Importer` not supported

Reply:
removed.